testing/mock.ts

```diff
@@ -643,15 +654,15 @@ function methodSpy<
   Return,
 >(self: Self, property: keyof Self): MethodSpy<Self, Args, Return> {
   if (typeof self[property] !== "function") {
-    throw new MockError("Property is not an instance method");
+    throw new TypeError("Property is not an instance method");
```

Comment:
This looks like trying an invalid spying. Why isn't this MockError?

Reply:
The property is the wrong type; it should be a function.

Reply:
The choice between TypeError and MockError seems arbitrary and confusing to me.

Reply:
Mocking fails because the property is the wrong type. There's overlap, but I feel to see what's confusing.